### PR TITLE
add reporting

### PR DESCRIPTION
**Pull request by SuddenGunter**

### CODE REVIEW

File: src/report/report.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	pipeline := []bson.M{
+		bson.M{
+			"$match": bson.M{"time": bson.D{
+				{"$gte", start},
+				{"$lt", end},
+			}},
+		},
+		bson.M{
+			"$match": bson.M{"category._id": bson.D{
+				{"$exists", "true"},
```
Comment (markelrep) [inline src/report/report.go]: true should be bool

File: src/report/report.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+				{"$ne", primitive.Null{}},
```
Comment (markelrep) [inline src/report/report.go]: "$ne", primitive.Null{} - maybe redundant ?

File: src/api/handlers/reports.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	err := c.BindQuery(&query)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, ResponseFromError(err))
+		return
+	}
+	reportResponse, err := handler.generator.GetReport(c, query.Start, query.End)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, ResponseFromError(err))
+		return
+	}
```
Comment (markelrep) [inline src/api/handlers/reports.go]: What will the behavior of the report handler if nothing found on a specified period?

**Comment by markelrep** [comment]
Fix all errors of go vet and golint

File: src/report/report.go (hunk 2 above)
Comment (SuddenGunter) [reply src/report/report.go]: what if we want to unset category, as you suggested earlier?

File: src/api/handlers/reports.go (hunk 3 above)
Comment (SuddenGunter) [reply src/api/handlers/reports.go]: api returned entity with field equals to null before, after https://gith ub.com/lungria/spendshelf-backend/pull/13/commits/bd2097f20684c56bebb2ec365af2ba29c573f8a2#diff-76434d2157e2ce5df37272308bd721daR68 
returns entity with empty array field |

```
{
    "report": []
}
```

File: src/report/report.go (hunk 2 above)
Comment (markelrep) [reply src/report/report.go]: Ok, I agree with this. It's maybe an empty object

File: src/api/handlers/reports.go (hunk 3 above)
Comment (markelrep) [reply src/api/handlers/reports.go]: Ok

**Pull request merged**